code/clue.js

```diff
@@ -10,50 +9,202 @@ const mrGreen = {
+  firstName: 'Cassandra',
+  lastName: 'Scarlet',
+  color: 'red',
+  description: 'She is very smart',
+  age: 37,
+  image: 'assets/scarlet.png',
+  occupation: 'Lawyer'
+}
+
+const mrsPeacock = {
+  firstName: 'Eleanor',
+  lastName: 'Peacock',
+  color: 'blue',
+  description: 'She is very rich',
+  age: 58,
+  image: 'assets/peacock.png',
+  occupation: 'Doctor'
+}
+
+const colonelMustard = {
+  firstName: 'Jack',
+  lastName: 'Mustard',
+  color: 'yellow',
+  description: 'He is very loyal',
+  age: 62,
+  image: 'assets/mustard.png',
+  occupation: 'Colonel'
+}
+
+const mrsWhite = {
+  firstName: 'Mrs',
+  lastName: 'White',
+  color: 'white',
+  description: 'She is very kind',
+  age: 65,
+  image: 'assets/white.png',
+  occupation: 'Actress'
+}
 
 const rope = {
   name: 'rope',
-  weight: 10
+  weight: 10,
+  color: 'black',
+  image: 'assets/rope.png'
+}
+
+const knife = {
+  name: 'knife',
+  weight: 4,
+  color: 'black',
+  image: 'assets/knife.png'
+}
+
+const candlestick = {
+  name: 'candestick',
+  weight: 8,
+  color: 'black',
+  image: 'assets/candlestick.png',
+}
+
+const dumbbell = {
+  name: 'dumbbell',
+  weight: 14,
+  color: 'black',
+  image: 'assets/dumbbell.png'
+}
+
+const poison = {
+  name: 'poison',
+  weight: 1,
+  color: 'black',
+  image: 'assets/poison.png'
 }
 
-// THE ROOMS ONLY HAS A NAME SO NO NEED FOR OBJECTS THERE.
+const axe = {
+  name: 'axe',
+  weight: 12,
+  color: 'black',
+  image: 'assets/axe.png'
+}
 
-// NOW GROUP ALL SUSPECTS, WEAPONS AND ROOMS IN ARRAYS LIKE THIS:
+const bat = {
+  name: 'bat',
+  weight: 9,
+  color: 'black',
+  image: 'assets/bat.png'
+}
+
+const trophy = {
+  name: 'trophy',
+  weight: 7,
+  color: 'black',
+  image: 'assets/trophy.png'
+}
+
+const pistol = {
+  name: 'pistol',
+  weight: 5,
+  color: 'black',
+  image: 'assets/pistol.png'
+}
 
 const suspects = [
   mrGreen,
-  mrsWhite
-  // ...  and the rest
+  mrsWhite,
+  mrsPeacock,
+  colonelMustard,
+  professorPlum,
+  missScarlet
+]
+
+const weapons = [
+  rope,
+  knife,
+  candlestick,
+  dumbbell,
+  poison,
+  axe,
+  bat,
+  trophy,
+  pistol
 ]
 
-const weapons = []
+const rooms = [
+  'Dining Room',
+  'Conservatory',
+  'Kitchen',
+  'Study',
+  'Library',
+  'Billiard Room',
+  'Lounge',
+  'Ballroom',
+  'Hall',
+  'Spa',
+  'Living Room',
+  'Observatory',
+  'Theater',
+  'Guest House',
+  'Patio'
+]
+
+console.log(suspects)
+console.log(weapons)
+console.log(rooms)
 
-const rooms = []
 
-// THIS FUNCTION WILL RANDOMLY SELECT ONE ITEM FROM THE ARRAY THAT YOU PASS IN TO THE FUNCTION.
-// YOU DON'T NEED TO CHANGE THIS, JUST TRY TO UNDERSTAND IT. AND HOW TO USE IT.
 const randomSelector = array => {
-  return array[Math.floor(Math.random() * array.length)]
+  const randomArrayIndex = Math.floor(Math.random() * array.length)
+  return array[randomArrayIndex]
 }
 
-// CREATE AN OBJECT THAT KEEPS THE MYSTERY.
-// With a killer, a weapon and a room.
-// The values will be set later.
+const mystery = {
```

Comment:
Excellent! 🌟 Having declared the mystery object values as undefined, the user is stopped from revealing the mystery before all the decks have been picked from. Otherwise, it would have shown: 'the murder was committed by undefined with an undefined in undefined.'

Reply:
It still does though! Will make an additional comment about this further down :)

PS. For more semantic code you could declare them with `null`